backend/main.py

```diff
@@ -0,0 +1,93 @@
+from fastapi import FastAPI, WebSocket
+from typing import Union
+from pydantic import BaseModel
+import json
+from threading import Thread
+import time
+
+from game_handler import GameHandler
+
+app = FastAPI()
+
+
+class ConnectionManager:
+    def __init__(self):
+        self.active_connections: list[WebSocket] = []
+
+    async def connect(self, websocket: WebSocket):
+        await websocket.accept()
+        self.active_connections.append(websocket)
+
+    def disconnect(self, websocket: WebSocket):
+        self.active_connections.remove(websocket)
+
+    async def send_personal_message(self, message: str, websocket: WebSocket):
+        await websocket.send_text(message)
+
+    async def broadcast(self, message: str):
+        for connection in self.active_connections:
+            await connection.send_text(message)
+
+
+manager = ConnectionManager()
+
+game_handler = GameHandler()
+game = game_handler.get_game()
+
+# Server loop
+def loop():
+    while True:
```

Comment:
This is gonna sit in a very tight loop doing nothing most of the time. We need to find a slightly better way of doing this.
In the case when no one is in the lobby (majority of time), we need some way of blocking the thread until a player joins.

In the case when a game is being played, we can probably get a way with doing 10 or 100 iterations per second.

Reply:
For now perhaps a sleep will suffice

Reply:
Used the threading module to get event-based handling for when there is no player.